packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -404,63 +493,261 @@ const handler: AdaptingProxyHandler<ProxyTarget, EditableTree> = {
+        // but it is a TypeError to return non-configurable for properties that do not exist on target,
+        // so they must return true.
+        if (typeof key === "symbol") {
+            switch (key) {
+                case proxyTargetSymbol:
+                    return {
+                        configurable: true,
+                        enumerable: false,
+                        value: target,
+                        writable: false,
+                    };
+                default:
+            }
+        } else {
+            if (key in { length: true, fieldKey: true, fieldSchema: true, primaryType: true }) {
+                return {
+                    configurable: true,
+                    enumerable: false,
+                    value: Reflect.get(target, key),
+                    writable: false,
+                };
+            } else if (keyIsValidIndex(key, target.length)) {
+                return {
+                    configurable: true,
+                    enumerable: true,
+                    value: target.proxifyNode(Number(key)),
+                    writable: false,
+                };
+            }
+        }
+        return undefined;
+    },
+};
+
 /**
  * See {@link UnwrappedEditableTree} for documentation on what unwrapping this performs.
  */
-function inProxyOrUnwrap(target: ProxyTarget, unwrap: boolean): UnwrappedEditableTree {
-    if (unwrap) {
-        const fieldSchema = target.getType(undefined, false) as NamedTreeSchema;
-        if (isPrimitive(fieldSchema)) {
+function inProxyOrUnwrap(
+    target: NodeProxyTarget | FieldProxyTarget,
+    unwrap: boolean,
+): UnwrappedEditableTree {
+    // Unwrap primitives or nodes having a primary field. Sequences unwrap nodes on their own.
+    if (unwrap && !isFieldProxyTarget(target)) {
+        const nodeType = target.getType(undefined, false) as NamedTreeSchema;
+        if (isPrimitive(nodeType)) {
             const nodeValue = target.cursor.value;
             if (isPrimitiveValue(nodeValue)) {
                 return nodeValue;
             }
             assert(
-                fieldSchema.value === ValueSchema.Serializable,
+                nodeType.value === ValueSchema.Serializable,
                 0x3c7 /* `undefined` values not allowed for primitive fields */,
             );
         }
         const primary = target.getPrimaryArrayKey();
         if (primary !== undefined) {
             target.cursor.enterField(primary.key);
-            const childTargets = mapCursorField(
+            const primarySequence = new FieldProxyTarget(
+                target.context,
+                primary.schema,
                 target.cursor,
-                (c) => new ProxyTarget(target.context, c),
+                nodeType.name,
             );
-            target.cursor.exitField();
-            return childTargets.map((childTarget) => inProxyOrUnwrap(childTarget, unwrap));
+            target.free();
```

Comment:
This makes the provided target unusable.

If we want to do this, it should be documented in the function documentation since someone might expect to be able to use this to unwrap a target that someone is actually using the proxy for (ex: if we had an unwrap function on EditiableTree, it looks like it could use this, but that would sometimes break).

If we move toward reusing the proxy targets/nodes, this is also likely to become a problem.

I think we might not want to free this here.

Reply:
Fixed.